src/Sprites/spriteMap.ts

```diff
@@ -0,0 +1,627 @@
+            data[i * 4 + 2 + (this.spriteCount * 8)] = r;
+            data[i * 4 + 3 + (this.spriteCount * 8)] = t ;
+        }
+
+        let floatArray = new Float32Array(data);
+
+        let t = RawTexture.CreateRGBATexture(
+        floatArray,
+        this.spriteCount,
+        4,
+        this._scene,
+        false,
+        false,
+        Texture.NEAREST_NEAREST,
+        Engine.TEXTURETYPE_FLOAT
+        );
+
+        return t;
+    }
+
+    /**
+    * Creates the tileMap texture Buffer
+    * @param buffer normaly and array of numbers, or a false to generate from scratch
+    * @param _layer indicates what layer for a logic trigger dealing with the baseTile.  The system uses this
+    * @returns RawTexture of the tileMap
+    */
+    createTileBuffer(buffer: any, _layer: number= 0): RawTexture {
+        let data = new Array();
+
+        if (!buffer) {
+            let bt = this.options.baseTile;
+            if (_layer != 0) {
+                bt = 0;
+            }
+            for (let y = 0; y < this.options.stageSize.y; y++) {
+                for (let x = 0; x < this.options.stageSize.x * 4; x += 4) {
+                    data.push(bt, 0, 0, 0);
+                }
+            }
+        }else {
+            data = buffer;
+        }
+
+        let floatArray = new Float32Array(data);
+        let t = RawTexture.CreateRGBATexture(
+        floatArray,
+        this.options.stageSize.x,
+        this.options.stageSize.y,
+        this._scene,
+        false,
+        false,
+        Texture.NEAREST_NEAREST,
+        Engine.TEXTURETYPE_FLOAT
+        );
+
+        return t;
+    }
+
+    /**
+    * Modfies the data of the tileMaps
+    * @param _layer is the ID of the layer you want to edit on the SpriteMap
+    * @param pos is the iVector2 Coordinates of the Tile
+    * @param tile The SpriteIndex of the new Tile
+    */
+    changeTiles(_layer: number= 0, pos: any , tile: number= 0): void {
+
+        let buffer: any = [];
+        buffer = this._tileMaps[_layer]!._texture!._bufferView;
+        if (!buffer) {
+            return;
+        }
+
+        let p = new Array();
+        if (pos instanceof Vector2) {
+            p.push(pos);
+        }else {
+            p = pos;
+        }
+
+        for (let i = 0; i < p.length; i++) {
+            let _p = p[i];
+            _p.x = Math.floor(_p.x);
+            _p.y = Math.floor(_p.y);
+            let id = (_p.x * 4) + (_p.y * (this.options.stageSize.x * 4));
+            buffer[id] = tile;
+        }
+
+        let t = this.createTileBuffer(buffer);
+        this._tileMaps[_layer].dispose();
+        this._tileMaps[_layer] = t;
+        this._material.setTextureArray("tileMap", this._tileMaps);
+    }
+
+    /**
+    * Creates the animationMap texture Buffer
+    * @param buffer normaly and array of numbers, or a false to generate from scratch
+    * @returns RawTexture of the animationMap
+    */
+    createTileAnimationBuffer(buffer: any): RawTexture {

```

Comment:
All the utilities function should be flag as private and not exposed publicly (keep the API simple)